vignettes/datatable-importing.Rmd

```diff
@@ -195,6 +195,88 @@ For more canonical documentation of defining packages dependency check the offic
+
+## How to convert your Depends dependency on data.table to Imports
+
+To convert a `Depends` dependency on `data.table` to an `Imports` dependency in an R package, follow these steps:
+
+### 1. Update the DESCRIPTION File
+In your package's `DESCRIPTION` file, change the entry from `Depends` to `Imports`. This involves moving `data.table` from the `Depends` field to the `Imports` field.
+
+**Before:**
+```r
+Depends:
+    R (>= 3.5.0),
+    data.table
+Imports:
+```
+
+**After:**
+```r
+Depends:
+    R (>= 3.5.0)
+Imports:
+    data.table
+```
+
+### 2. Modify the NAMESPACE File
+Ensure that any functions from `data.table` that you use in your package are imported. This can be done either by explicitly importing specific functions or by importing the entire package but specific imports are considered best practice.
+
+#### Importing Specific Functions
+If you use specific functions like `data.table()`, `setDT()`, or `merge()`, you can import them individually in the `NAMESPACE` file:
+
+```r
+importFrom(data.table, data.table, setDT, merge)
+```
+
+#### Bulk Import
+Alternatively, you can import all functions from `data.table` at once, though this is generally not recommended as best practice:
+```r
+import(data.table)
+```
+
+By including these import statements in your `NAMESPACE` file, you ensure that the necessary functions are available for your package to use.
+
+### 3. Update Your R code files
+When you move a package from `Depends` to `Imports`, it will no longer be automatically attached when your package is loaded. This means you need to refer to `data.table` functions explicitly using the `data.table::` prefix, or import the functions in your scripts. However, if you import `data.table` in your Namespace, you won't need to make any changes in your R scripts.
+
+**Before (with `Depends`):**
+```r
+# data.table functions are directly available
+dt <- data.table(x = 1:10, y = letters[1:10])
+setDT(dt)
+result <- merge(dt, other_dt, by = "x")
+```
+
+**After (with `Imports`):**
+```r
+# Use the data.table:: prefix or import functions explicitly
+dt <- data.table::data.table(x = 1:10, y = letters[1:10])
+data.table::setDT(dt)
+result <- data.table::merge(dt, other_dt, by = "x")
+```
+
+Alternatively, you can import specific functions at the top of your R scripts:
+
+```r
+# Import specific functions at the beginning of the script
+data.table::data.table
+data.table::setDT
+data.table::merge
+
+# Now you can use them without the data.table:: prefix
+dt <- data.table(x = 1:10, y = letters[1:10])
+setDT(dt)
+result <- merge(dt, other_dt, by = "x")
+```
+
+**Important**: To avoid the need for changes in your R scripts, ensure you import `data.table` in your Namespace. This allows the functions to be available as they were with `Depends`.
+
+### Benefits of Using `Imports`
+- **Namespace Management**: Only the functions your package explicitly imports are available, reducing the risk of function name clashes.
```

Comment:
It is also much easier for maintenance, as mentioned also above as your upstream dependencies' own APIs evolve, relying too much on `Depends` is bound to cause headaches & conflicts.